File: BinarySearch.py
def bisearch(arr, low, high, x):
 
    
    if high >= low:
 
        mid = (high + low) // 2
        if arr[mid] == x:
            return mid
 
       
        elif arr[mid] > x:
            return bisearch(arr, low, mid - 1, x)
 
        
        else:
            return bisearch(arr, mid + 1, high, x)
 
    else:
      
        return -1
 

arr = [ 2, 3,68,24,90]
x = 10
 

result = bisearch(arr, 0, len(arr)-1, x)
 
if result != -1:
    print("Element is present at index", str(result))
else:
    print("Element is not present in array")

File: Binary Search.c
#include<stdio.h>
int binarysearch(int a[],int low,int high,int x)
{
    if(high>=low)
    {
     int mid=low+(high-low)/2;
     if(a[mid]==x)
     return mid;
     else if(a[mid]>x)
     return  binarysearch(a,low,mid-1,x);
     else
     return  binarysearch(a,mid+1,high,x);
   }
     else
     return -1;
}
int main(void)
{
    int a[10]={10,15,30,55,70,80,250};
    int n=sizeof(a)/sizeof(a[0]);
    int x;
    printf("enter search element:\n");
    scanf("%d",&x);
    int result= binarysearch(a,0,n-1,x);
    (result==-1)?printf("element is not present in array"):printf("element is present at index %d",result);
    return 0;  
}